
/NodeRgbdAlignment:
  ros__parameters:
    tf:
      base_link_id: world
      frame_id: odom
    odometry:
      method: icp
      methods: [rgbd, icp]
      includeKeyFrame: false
      trackKeyFrame: false
      rgbd:
        includePrior: false
        initOnPrior: false
        features:
          min_gradients: [0.0,0.0,0.0,0.0]
          min_depth: 0.5
          max_depth: 4.0
          max_points_part: [1.0,1.0,1.0,1.0]
        loss:
          function: None
          functions: [None, Huber, tdistribution]
          huber:
            c: 1.43
          tdistribution:
            v: 5.0
        pyramid:
          levels: [1.0, 0.5, 0.25, 0.125]
        solver:
          max_iterations: 50
          min_step_size: 0.0001
          min_gradient: 0.0000000001
          max_increase: 1.1
          
    prediction:
      model: NoMotion
      models: [NoMotion, ConstantMotion, Kalman]
      constant_motion:
        covariance: [0.01,0.01,0.01,0.01,0.01,0.01]
      kalman: 
      # uncertainties are 1 std errors in SI units
        initial_uncertainty:
          pose:
            translation: 0.0001 # [m]
            rotation: 0.0001 # [째]
          velocity:
            translation: 0.00001 # [m/s]
            rotation: 0.001 # [째/s]
        process_noise: # Q matrix
          pose:
            translation: 0.001 # [m]
            rotation: 0.0001 # [째]
          velocity:
            translation: 0.0001 # [m/s]
            rotation: 0.001 # [째/s]
    keyframe_selection:
      method: custom
      methods: [custom, idx, never]
      idx:
        period: 5
      custom:
        max_translation: 0.1
        max_rotation: 1.0
        min_visible_points: 0
    map:
      n_keyframes: 5
      n_frames: 2
    bundle_adjustment:
      max_iterations: 0
      huber_constant: 3.0
    log:
      config_dir: /home/ros/vslam_ros/install/vslam_ros/share/vslam_ros/config/log
      # root_dir: /home/ros/vslam_ros/log/run
      image:
        ImageWarped:
          show: false
          block: false
        Residual:
          show: false
          block: false
        Weights:
          show: false
          block: false
        Image:
          show: false
          block: false
        Template:
          show: false
          block: false
        Depth:
          show: false
          block: false
        Tracking:
          show: false
          block: false
        KeyFrames:
          show: false
          block: false
          save: false
        ErrorDistribution:
          show: false
          block: false
          save: false
        Kalman:
          show: false
          block: false
          save: true
          rate: 1
        Solver:
          show: false
          block: false
          save: false
          rate: 1
        Alignment:
          show: false
          block: false
          save: true
          rate: 1  
    qos_overrides:
      /parameter_events:
        publisher:
          depth: 1000
          durability: volatile
          history: keep_last
          reliability: reliable
      /tf:
        publisher:
          depth: 100
          durability: volatile
          history: keep_last
          reliability: reliable
#/NodeResultWriter:
#/NodeGtLoader:
/NodeReplayer:
  ros__parameters:
    delay: 0.0
    duration: 20.0
    timeout: 1.0

